utils/techutils/techutils.go

```diff
@@ -340,12 +342,68 @@ func DetectTechnologiesDescriptors(path string, recursive bool, requestedTechs [
 		log.Debug(fmt.Sprintf("mapped %d working directories with indicators/descriptors:\n%s", len(workingDirectoryToIndicators), strJson))
 	}
 	technologiesDetected, err = mapWorkingDirectoriesToTechnologies(workingDirectoryToIndicators, excludedTechAtWorkingDir, ToTechnologies(requestedTechs), requestedDescriptors)
+	if err != nil {
+		return
+	}
+	if recursive {
+		// If recursive search, we need to also make sure to include directories that do not have any technology indicators.
+		technologiesDetected, err = addNoTechIfNeeded(technologiesDetected, path, excludePathPattern)
+	}
+	// In case we have recursive search, and the
 	if len(technologiesDetected) > 0 {
 		log.Debug(fmt.Sprintf("Detected %d technologies at %s: %s.", len(technologiesDetected), path, maps.Keys(technologiesDetected)))
 	}
 	return
 }
 
+func addNoTechIfNeeded(technologiesDetected map[Technology]map[string][]string, path, excludePathPattern string) (finalMap map[Technology]map[string][]string, err error) {
```

Comment:
I would like to see unit_tests for the new functions

Reply:
added at https://github.com/jfrog/jfrog-cli-security/pull/221